File: functions/registry.js
export const onRequest = async (context) => {
  const url = new URL(context.request.url);
  
  if (url.pathname === "/registry" || url.pathname === "/registry.html") {
    return Response.redirect("https://www.myregistry.com/giftlist/ethan-and-tiffany", 301);
  }
  return new Response("Not Found", { status: 404 });
};

File: functions/registry.py
async def on_request(request):
    """
    Handle requests for /registry and /registry.html by redirecting to another URL.
    """
    path = request.url.path

    if path in ["/registry", "/registry.html"]:
        return {
            "status": 301,
            "headers": {
                "Location": "https://www.myregistry.com/giftlist/ethan-and-tiffany"  # Replace with your target URL
            }
        }

    # Optional: Return a 404 for other paths not matched
    return {
        "status": 404,
        "body": "Not Found"
    }

